json/json_parser.go

```diff
@@ -46,6 +46,21 @@ func ParseJsonHash(path string) (map[string]interface{}, error) {
 	return stringMap, nil
 }
 
+func ParseJsonFromFileOrString(fileOrJson string) (map[string]interface{}, error) {
+	var jsonMap map[string]interface{}
+	var err error
+
+	jsonMap, err = ParseJsonHash(fileOrJson)
+	if err != nil {
```

Comment:
The error message is hidden from the user in case of the file contains invalid string, because it is over-written by the error message at line 56. If it was actually a file containing invalid json, the error message explaining that will never get displayed to the user. They'll see an error message about how their filename was invalid json instead. There's a go command to validate a filepath that you can use to check if it's a string or filepath, and switch from there.